include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -1176,6 +1176,26 @@ def DoubleConstantOp : FIRRTLOp<"double", [Pure, ConstantLike]> {
   let assemblyFormat = "$value attr-dict";
 }
 
+class IntegerBinaryPrimOp<string mnemonic, list<Trait> traits = []> :
+   BinaryPrimOp<mnemonic, FIntegerType, FIntegerType, FIntegerType,
+                [Pure] # traits> {
+  let inferType = "impl::inferIntegerBinaryPrimResult";
+}
+
+def IntegerAddOp : IntegerBinaryPrimOp<"integer.add", [Commutative]> {
+  let summary = "Add two FIntegerType values";
+  let description = [{
+    The add operation result is the arbitrary precision signed integer
+    arithmetic sum of `lhs` and `rhs`.
```

Comment:
I was thinking this would make more sense in the rendered docs online where it would define `lhs` and `rhs`, like for other BinaryPrimOps: https://circt.llvm.org/docs/Dialects/FIRRTL/#firrtladd-circtfirrtladdprimop. But I will just change this to read "sum of the two operands".